py/dml/structure.py

```diff
@@ -2394,6 +2437,173 @@ def mkexpr(self, indices):
         return mkList(self.site,
                       [param.mkexpr(indices) for param in self.params])
 
+def port_class_ident(port):
+    '''The C identifier used for a port class within the device class
+    initialization function'''
+    return ('class' if port.objtype == 'device'
+            else '_port_class_' + port.attrname())
+
+def get_attr_name(prefix, node):
+    if node.objtype == 'register' and node.is_confidential():
+        return ctree.get_anonymized_name(node)
+    else:
+        return prefix + node.name
+
+
+class ConfAttrNameParamExpr(objects.ParamExpr):
+    '''The _attr_name parameter of an _conf_attribute object'''
+    __slots__ = ('site', 'node', 'cached',)
+
+    def __init__(self, site, node):
+        self.site = site
+        self.node = node
+        self.cached = None
+
+    def mkexpr(self, indices):
+        if self.cached is not None:
+            return self.cached
+        parent = self.node.parent
+        prefix = ""
+
+        while parent.objtype not in ('device', 'bank', 'port', 'subdevice'):
+            prefix = parent.name + "_" + prefix
+            parent = parent.parent
+
+        self.cached = mkStringConstant(self.site,
+                                       get_attr_name(prefix, self.node))
```

Comment:
This is 1.4 only, where anonymization is softcoded, so no need for transformation.

Reply:
I guess not... but if `get_attr_name` is supposed to be the canonical way to retrieve the C name for an attribute, circumventing it is Bad and Unidiomatic.